### PR TITLE
Insights playbook.

### PR DESCRIPTION
**Pull request by dcadams**

### CODE REVIEW

**Comment by dcadams** [comment]
Associated secure pr:
https://github.com/Stanford-Online/configuration-secure/pull/330

**Comment by dcadams** [comment]
@stvstnfrd whenever you have a few minutes.

File: stanford/playbooks/insights.yml [hunk 1]
```diff
@@ -1,16 +1,39 @@
...
+  become: True
   vars:
-    COMMON_ENABLE_DATADOG: False
-    COMMON_ENABLE_SPLUNKFORWARDER: False
+    CLUSTER_NAME: insights
+    migrate_db: "yes"
+    disable_edx_services: false
+    ENABLE_DATADOG: False
+    ENABLE_SPLUNKFORWARDER: False
+    ENABLE_NEWRELIC: False
```
Comment (stvstnfrd) [inline stanford/playbooks/insights.yml]: These three `ENABLE_*` variables already default to `False`, so we should omit them here.

File: stanford/playbooks/insights.yml [hunk 2]
```diff
@@ -1,16 +1,39 @@
...
+  hosts: "tag_Cluster_{{ CLUSTER_NAME }}:&tag_todo_deploy:&tag_Deployment_{{ COMMON_DEPLOYMENT }}:&tag_Number_{{ CLUSTER_NUMBER }}"
   gather_facts: True
-  vars_files:
-    - "{{ COMMON_DEPLOYMENT_DIR }}/insights.yml"
+  become: True
   vars:
-    COMMON_ENABLE_DATADOG: False
-    COMMON_ENABLE_SPLUNKFORWARDER: False
+    CLUSTER_NAME: insights
+    migrate_db: "yes"
```
Comment (stvstnfrd) [inline stanford/playbooks/insights.yml]: Nit: single quotes for non-interpolated strings.

File: stanford/playbooks/insights.yml [hunk 3]
```diff
@@ -1,16 +1,39 @@
...
     - insights
-    - role: splunkforwarder
-      when: COMMON_ENABLE_SPLUNKFORWARDER
+    - role: postfix_queue
+      when: POSTFIX_QUEUE_EXTERNAL_SMTP_HOST != ''
+  tasks:
+    - name: 'Install libffi-dev'
+      apt: name=libffi-dev state=present
+    - name: 'Install make'
+      apt: name=make state=present
```
[the remark this answers is not in the document]
Comment (dcadams) [reply stanford/playbooks/insights.yml]: Yes, not in the insights role. Also, not in the edx insights playbook but that is in the edx-east area.

However, it is in the opencraft plays so I assume it's required although I can't tell you what will happen if it isn't included.

File: stanford/playbooks/insights.yml [hunk 4]
```diff
@@ -1,16 +1,39 @@
...
-    COMMON_ENABLE_SPLUNKFORWARDER: False
+    CLUSTER_NAME: insights
+    migrate_db: "yes"
+    disable_edx_services: false
+    ENABLE_DATADOG: False
+    ENABLE_SPLUNKFORWARDER: False
+    ENABLE_NEWRELIC: False
+  vars_files:
+    - "{{ COMMON_DEPLOYMENT_DIR }}/common.yaml"
+    - "{{ COMMON_DEPLOYMENT_DIR }}/analytics-api.yml"
```
[the remark this answers is not in the document]
Comment (dcadams) [reply stanford/playbooks/insights.yml]: I think it's just the 'Common Vars' section.

File: stanford/playbooks/insights.yml [hunk 5]
```diff
@@ -1,16 +1,36 @@
...
+  become: True
   vars:
-    COMMON_ENABLE_DATADOG: False
-    COMMON_ENABLE_SPLUNKFORWARDER: False
+    CLUSTER_NAME: insights
+    migrate_db: 'yes'
+    disable_edx_services: false
+  vars_files:
+    - "{{ COMMON_DEPLOYMENT_DIR }}/common.yaml"
+    - "{{ COMMON_DEPLOYMENT_DIR }}/analytics-api.yml"
```
Comment (stvstnfrd) [inline stanford/playbooks/insights.yml]: With the refactoring we've made to the secrets, I think this role no longer needs to include `analytics-api.yml`, right?
There should be one reference to an `ANALTYICS_API_*` var, but it's being loaded from `common.yaml`.

**Comment by stvstnfrd** [comment]
@dcadams I assume you're going to squash these commits together and reword?

**Pull request merged**